src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -327,15 +326,15 @@ private async Task<OpenApiOperation> GetOperationAsync(
         => description.ActionDescriptor.AttributeRouteInfo?.Name ??
             description.ActionDescriptor.EndpointMetadata.OfType<IEndpointNameMetadata>().LastOrDefault()?.EndpointName;
 
-    private static List<OpenApiTagReference> GetTags(ApiDescription description, OpenApiDocument document)
+    private static ISet<OpenApiTagReference> GetTags(ApiDescription description, OpenApiDocument document)
```

Comment:
extreme nit: Generally for private/internal methods we can return the concrete type instead of an interface, this allows for optimizations at the usage site. I know the return values here are just set on an OpenAPI type, but just a general pattern.